### PR TITLE
Issue #31: Narrow cleanup support compilation

### LINKED ISSUE #31: chore(tests): consolidate per-item #[allow(dead_code)] in tests/support/cleanup.rs
## Background

`tests/support/cleanup.rs` carries five separate `#[allow(dead_code)]` attributes (identified during PR #23 review, https://github.com/leynos/axinite/pull/23):

| Line | Item |
|------|------|
| 5 | `enum PathKind` |
| 12 | `pub struct CleanupGuard` |
| 17 | `impl CleanupGuard` block |
| 52 | `pub fn setup_test_dir` |
| 62 | `pub fn setup_test_dir_with_suffix` |

These are required because the file is compiled as part of the shared `support` module included in every test harness; harnesses that do not exercise filesystem cleanup never reference these items, triggering dead-code warnings.

## Proposed Change

Replace the five per-item `#[allow(dead_code)]` attributes with a single module-level `#![allow(dead_code)]` declaration at the top of `cleanup.rs`. This makes the intent (shared utility module, not every item is used by every harness) immediately clear and removes the repetition.

If any item in the file is genuinely unused across _all_ harnesses, remove it rather than suppressing the lint.

## Files Affected

- `tests/support/cleanup.rs` — lines 5, 12, 17, 52, 62

## References

- PR #23: https://github.com/leynos/axinite/pull/23

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- narrow test support roots so cleanup helpers compile only in harnesses that use them
- split the cleanup guard from directory setup helpers and remove stale cleanup symbol anchors
- keep the remaining support surfaces lint-clean under the narrower harness boundaries

Closes #31

## Summary by Sourcery

Narrow and reorganize test support modules so that cleanup, trace, webhook, and rig helpers compile only in the harnesses that use them, while adding coverage for trace utilities and keeping test support surfaces minimal and lint-clean.

New Features:
- Introduce harness-specific support entrypoints for support_unit_tests, e2e_traces, tools_and_config, channels, infrastructure, webhook-focused tests, and trace_llm tests.
- Add JSON patching helpers for tools_and_config trace tests and new trace construction helpers in trace_format tests.

Enhancements:
- Split the CleanupGuard RAII helper into its own module and re-export it only where needed, removing global symbol anchors from the shared support module.
- Extract shared webhook health check and HTTP client helpers into a reusable webhook_common module used by multiple harnesses.
- Tighten trace_llm and trace_types re-exports so only the required APIs are available per harness, reducing unnecessary coupling between tests.

Tests:
- Move load_trace_with_mutation tests into a dedicated trace_types_tests module under support_unit_tests and expand them to assert more properties.
- Add new tests validating file-based trace loading, path patching, and LlmTrace construction in tools_and_config trace_format tests.

### CODE REVIEW

File: tests/infrastructure.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 //! Infrastructure integration tests covering heartbeat, pairing, provider
 //! chaos, SIGHUP reload, and workspace functionality.
 
+#[path = "support/infrastructure.rs"]
 mod support;
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/infrastructure.rs:5]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/support_unit.rs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+#[cfg(feature = "libsql")]
+const _: fn(&test_rig::TestRig) -> Option<&std::sync::Arc<trace_llm::TraceLlm>> =
+    test_rig::TestRig::trace_llm;
+#[cfg(feature = "libsql")]
+const _: fn(test_rig::TestRigBuilder) -> AsyncBuildRig = _build_sig;
+#[cfg(feature = "libsql")]
+const _: for<'a> fn(&'a str) -> AsyncUnit<'a> = _run_recorded_trace_sig;
+const _: fn(&mut trace_types::LlmTrace, &str, &str) -> usize = trace_types::LlmTrace::patch_path;
+const _: for<'a> fn(&'a trace_types::LlmTrace) -> Vec<&'a ironclaw::llm::recording::TraceStep> =
+    trace_types::LlmTrace::playable_steps;
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/support_unit.rs:239]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/developers-guide.md [hunk 3]
```diff
@@ -174,6 +174,31 @@ The current `Makefile` also includes:
...
+separate from the broader `tests/support/` graph means the harness only
+compiles the helpers it actually needs, which keeps the support surface
+honest and avoids dead-code lint suppression.
+
+The `trace_llm` façade in that module provides a narrowed one-import
+surface for LLM trace helpers. It re-exports
+`trace_json_patch::patch_json_value` together with
+`trace_types::{LlmTrace, TraceExpects}` so tests can create trace
+fixtures, patch recorded JSON payloads, and assert on trace contents
+without reaching into the underlying support-module layout.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/mod.rs [hunk 4]
```diff
@@ -78,26 +78,23 @@ const _: fn() -> anyhow::Result<std::sync::Arc<ironclaw::tools::wasm::WasmToolRu
...
 
     touch!(
-        cleanup::CleanupGuard::new as fn() -> cleanup::CleanupGuard,
-        cleanup::setup_test_dir as fn(&str) -> std::io::Result<()>,
-        cleanup::setup_test_dir_with_suffix
-            as fn(&std::path::Path, &str) -> std::io::Result<String>,
+        fixtures::FIXTURE_ROOT,
+        fixtures::DEFAULT_TIMEOUT,
+        fixtures::LONG_TIMEOUT,
+        fixtures::fixture_path as fn(&str, &str) -> String,
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/mod.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/mod.rs [hunk 5]
```diff
@@ -376,7 +373,14 @@ fn touch_test_rig_symbols() {
...
 
 fn test_rig_symbol_refs() {
+<<<<<<< ours
+    touch_cleanup_symbols();
+||||||| original
     touch_cleanup_symbols();
+    touch_fixture_symbols();
+=======
+    touch_fixture_symbols();
+>>>>>>> theirs
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/mod.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/support_unit.rs [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+type AsyncCompletedToolCalls<'a> =
+    std::pin::Pin<Box<dyn std::future::Future<Output = Vec<(String, bool)>> + 'a>>;
+#[cfg(feature = "libsql")]
+type AsyncStatusEvents<'a> = std::pin::Pin<
+    Box<dyn std::future::Future<Output = Vec<ironclaw::channels::StatusUpdate>> + 'a>,
+>;
+#[cfg(feature = "libsql")]
+type AsyncTraceRun<'a> = std::pin::Pin<
+    Box<dyn std::future::Future<Output = Vec<Vec<ironclaw::channels::OutgoingResponse>>> + 'a>,
+>;
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/support_unit.rs:45]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/cleanup_guard.rs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        for (path, kind) in &self.paths {
+            match kind {
+                PathKind::File => {
+                    let _ = std::fs::remove_file(path);
+                }
+                PathKind::Dir => {
+                    let _ = std::fs::remove_dir_all(path);
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/cleanup_guard.rs:63]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/trace_llm.rs [hunk 8]
```diff
@@ -1,33 +1,6 @@
...
-        }
-        serde_json::Value::Object(obj) => {
-            for (_, v) in obj {
-                patch_json_value(v, from, to);
-            }
-        }
-        _ => {}
-    }
-}
+pub use super::trace_types::{LlmTrace, TraceExpects};
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/trace_llm.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### tests/support/trace_llm.rs (1)

`3-13`: _⚠️ Potential issue_ | _🟠 Major_
**Eliminate the `unused_imports` suppression from this shared façade.**
This narrowed surface still depends on `#[expect(unused_imports)]` for the recording-type re-export block. Move those exports into the harness-specific support roots that actually consume them, or anchor them through the symbol-ref pattern instead of keeping the suppression in `tests/support/trace_llm.rs`.
Based on learnings, "In leynos/axinite integration test support modules under tests/support/, avoid using #[allow(dead_code)] and #[allow(unused_imports)] on helper code ... only #[expect(dead_code, reason = '...')] is allowed as a last resort on the *_symbol_refs anchor function itself."

##### 🤖 Prompt for AI Agents

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### tests/support/trace_llm.rs (1)

`15-33`: _🧹 Nitpick_ | _🔵 Trivial_
**Consolidate `patch_json_value` to eliminate duplication.**
This function is duplicated in `tests/support/tools_and_config.rs` (lines 11–28). Extract it to a shared location (e.g., keep it here with `pub(crate)` visibility) and import it where needed.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### docs/developers-guide.md (1)

`1355-1372`: _⚠️ Potential issue_ | _🟡 Minor_
**Use `plaintext` for the staged-lifecycle fenced block.**
Change the code fence language from `text` to `plaintext` for this non-code

diagram block.
Triage: `[type:syntax/md]` `[type:docstyle]`

##### 📘 Proposed fix

```diff
-```text
+```plaintext
 SkillInstallPayload          (Markdown | DownloadedBytes)
         │
         ▼
 prepare_install_to_disk()   creates .<uuid>/ under install_root,
@@
   atomic rename staged→final
   in-memory registry insert
```

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### tests/support/trace_llm.rs (1)

`3-13`: _⚠️ Potential issue_ | _🟠 Major_
**Eliminate the `unused_imports` suppression from this shared façade.**
This narrowed surface still depends on `#[expect(unused_imports)]` for the recording-type re-export block. Move those exports into the harness-specific support roots that actually consume them, or anchor them through the symbol-ref pattern instead of keeping the suppression in `tests/support/trace_llm.rs`.
Based on learnings, "In leynos/axinite integration test support modules under tests/support/, avoid using #[allow(dead_code)] and #[allow(unused_imports)] on helper code ... only #[expect(dead_code, reason = '...')] is allowed as a last resort on the *_symbol_refs anchor function itself."

##### 🤖 Prompt for AI Agents

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

### Assessment against linked issues

| Issue | Objective | Addressed | Explanation |
| --- | --- | --- | --- |
| [#31](https://github.com/leynos/axinite/issues/31) | Replace the five per-item `#[allow(dead_code)]` attributes in `tests/support/cleanup.rs` with a single module-level `#![allow(dead_code)]` declaration at the top of that file. | ❌ | The PR removes the `CleanupGuard` implementation from `tests/support/cleanup.rs` into a new `tests/support/cleanup_guard.rs` module and restructures which harnesses compile which support modules. It does not add a module-level `#![allow(dead_code)]` to `cleanup.rs` (or the new `cleanup_guard.rs`), so the specific requested consolidation of per-item `#[allow(dead_code)]` attributes into one module-level allow is not implemented. |
| [#31](https://github.com/leynos/axinite/issues/31) | Ensure that any items in `tests/support/cleanup.rs` that are genuinely unused across all test harnesses are removed rather than simply having their dead-code warnings suppressed. | ✅ |

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

### ❌ Failed checks (2 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Developer Documentation | ⚠️ Warning | PR introduces major architectural changes (six harness-specific support roots, CleanupGuard restructuring, narrowed test compilation scopes) without corresponding developer guide documentation, leaving the pattern and implementation details undocumented. | Document all six harness-specific support roots, explain CleanupGuard restructuring, clarify how the pattern eliminates dead-code lint suppressions, and address unresolved review comments (PathBuf refactoring, spelling correction, line 226 wording). |
| Module-Level Documentation | ⚠️ Warning | Four public functions lack required documentation: CleanupGuard::new(), CleanupGuard::file(), CleanupGuard::dir(), and setup\_test\_dir(). | Add doc comments to all four public functions documenting their purpose, parameters, return types, and method chaining support per Rust documentation standards. |

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational.

### ❌ Failed checks (2 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Developer Documentation | ⚠️ Warning | Pull request introduces six harness-specific support modules and CleanupGuard restructuring without adequate documentation of architectural entry points and dead-code suppression elimination pattern. | Document five undocumented harness-specific support modules, explain CleanupGuard restructuring and compilation scope narrowing, expand LLM acronym on first use in section 11, and update section 10 guidance. |
| Module-Level Documentation | ⚠️ Warning | The public function `setup_test_dir` in `tests/support/cleanup.rs` lacks a documentation comment, whilst `setup_test_dir_with_suffix` includes one. | Add a doc comment to `setup_test_dir` explaining its purpose of removing and recreating a test directory, following the pattern in `setup_test_dir_with_suffix`. |

File: docs/developers-guide.md [hunk 9]
```diff
@@ -218,7 +218,37 @@ use, then move it into `tests/support/mod.rs`. Do not promote
...
+
+`tests/support/tools_and_config.rs` is a harness-specific support root
+for the `tools_and_config` integration binary. Keeping that entrypoint
+separate from the broader `tests/support/` graph means the harness only
+compiles the helpers it requires, which keeps the support surface
+honest and avoids dead-code lint suppression.
+
+The `trace_llm` facade in that module provides a narrowed one-import
+surface for LLM trace helpers. It exports
+`trace_types::{LlmTrace, TraceExpects}` publicly for test consumers,
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md:254]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: Makefile [hunk 10]
```diff
@@ -38,6 +39,14 @@ lint:
 	$(CARGO) clippy --all --benches --tests --examples --all-features $(TEST_FEATURES) -- -D warnings
 	$(CARGO) clippy --manifest-path $(GITHUB_TOOL_MANIFEST) --tests -- -D warnings
 
+markdownlint:
+	@files="$$(git diff --name-only origin/main...HEAD -- '*.md')"; \
+	if [ -n "$$files" ]; then \
+		$(BUNX) markdownlint-cli2 $$files; \
+	else \
+		echo "No changed Markdown files to lint."; \
+	fi
```
[the remark this answers is not in the document]
Comment (leynos) [reply Makefile]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/support/test_rig/mod.rs [hunk 11]
```diff
@@ -6,11 +6,9 @@
 
 mod builder;
 mod channel_handle;
-mod helpers;
+pub(crate) mod helpers;
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/test_rig/mod.rs:9]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
### ❌ Failed checks (2 errors)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | PR introduces new functionality in webhook\_common.rs and trace\_json\_patch.rs without substantive unit tests covering actual behaviour. | Write focused unit tests for webhook\_common.rs functions verifying endpoint registration and timeout application; add unit tests for patch\_json\_value() covering edge cases and recursion; remove vacuous trybuild instantiation tests lacking output assertions. |
| Module-Level Documentation | ❌ Error | Multiple harness-specific support root modules expose public declarations without individual doc comments, violating documentation standards. | Add individual doc comments (///) above each public module declaration in support roots to clarify functionality and harness-specific composition. |
